### DISCUSSION TITLE
Packet buffer lifetime

### DISCUSSION THREAD

**Discussion by CMoore-Darwinium** [libpnet/libpnet#517] [Ideas]
Hi everyone,

I was using libpnet and I ran into a small pain point.

I was iterating through Ipv6 headers in a loop and found it frustrating that the lifecycle of payload() return value is tied to that of the packet, not the underlying buffer.

I was wondering if there would be anything wrong with giving Packet a lifecycle:
```
pub trait Packet<'l> {
    /// Retrieve the underlying buffer for the packet.
    fn packet(&self) -> &'l [u8];

    /// Retrieve the payload for the packet.
    fn payload(&self) -> &'l [u8];
}
```
I'd probably attempt to write a patch, unless this sort of thing is considered undesirable, or it has already been determined to be impossible.

**Comment by infosechoudini** [comment]
ah, that's super frustrating, how did you fix it? did you clone the data into it's own buffer?
